tracer/test/Datadog.Trace.ClrProfiler.Managed.Tests/AutoInstrumentation/AWS/Kinesis/ContextPropagationTests.cs

```diff
@@ -0,0 +1,155 @@
+        data.Should().Be("Jordan Gonzalez");
+    }
+
+    [Fact]
+    public void InjectTraceIntoData_WithJsonString_AddsTraceContext()
+    {
+        var request = new PutRecordRequest { StreamName = StreamName, Data = ContextPropagation.DictionaryToMemoryStream(PersonDictionary) };
+
+        var proxy = request.DuckCast<IPutRecordRequest>();
+
+        ContextPropagation.InjectTraceIntoData<PutRecordsRequest>(proxy, spanContext);
+
+        // MemoryStreamToDictionary returns a Dictionary<string, object>
+        var dataDictionary = ContextPropagation.MemoryStreamToDictionary(proxy.Data);
+        var extracted = dataDictionary.TryGetValue(DatadogKey, out var datadogDictionary);
+        extracted.Should().BeTrue();
+
+        // Cast into a Dictionary<string, string> so we can extract it properly
+        var dictionary = JsonConvert.DeserializeObject<Dictionary<string, string>>(datadogDictionary.ToString());
+
+        var extractedSpanContext = SpanContextPropagator.Instance.Extract(dictionary);
+        extractedSpanContext.Should().NotBeNull();
+        extractedSpanContext.TraceId.Should().Be(spanContext.TraceId);
+        extractedSpanContext.SpanId.Should().Be(spanContext.SpanId);
+    }
+
+    [Theory]
+    [MemberData(nameof(MemoryStreamToDictionaryExpectedData))]
+    public void ParseDataObject_ReturnsExpectedValue(byte[] bytes, Dictionary<string, object> expected)
+    {
+        var memoryStream = new MemoryStream(bytes);
+        var result = ContextPropagation.ParseDataObject(memoryStream);
+        result.Should().BeEquivalentTo(expected);
+    }
+
+    [Fact]
+    public void MemoryStreamToDictionary_WithJsonString_ReturnsDictionary()
+    {
+        // JSON string
+        var personMemoryStream = new MemoryStream(PersonJsonStringBytes);
+        var personDictionary = ContextPropagation.MemoryStreamToDictionary(personMemoryStream);
+        personDictionary.Should().BeEquivalentTo(PersonDictionary);
+
+        // Base64 JSON string
+        var encodedPersonMemoryStream = new MemoryStream(PersonBase64StringBytes);
+        personDictionary = ContextPropagation.MemoryStreamToDictionary(encodedPersonMemoryStream);
+        personDictionary.Should().BeEquivalentTo(PersonDictionary);
+    }
+
+    [Fact]
+    public void MemoryStreamToDictionary_WithNonJsonString_ThrowsException()
+    {
+        // Anything that is not a JSON string, will throw an error
+        var streamNameMemoryStream = new MemoryStream(StreamNameBytes);
+        Assert.ThrowsAny<Exception>(() => ContextPropagation.MemoryStreamToDictionary(streamNameMemoryStream));
+    }
+
+    [Fact]
+    public void DictionaryToMemoryStream_ReturnsMemoryStream()
+    {
+        var personMemoryStream = ContextPropagation.DictionaryToMemoryStream(PersonDictionary);
+        personMemoryStream.Should().NotBeNull();
+
+        ContextPropagation.MemoryStreamToDictionary(personMemoryStream).Should().BeEquivalentTo(PersonDictionary);
```

Comment:
Similar to the other feedback I left in this file, I think this method would be stronger if it only tested `ContextPropagation.DictionaryToMemoryStream`, so instead of getting back a dictionary representation by calling `ContextPropagation.MemoryStreamToDictionary` I suggest using the Newtonsoft JSON library (which is pretty well-tested) to deserialize and assert our expectations.

As a result, if there's a problem with going from a dictionary to a memory stream, this test will point it out. If it's the other way around, then one of the `MemoryStreamToDictionary_*` test methods will point it out.

Reply:
Updated to compare bytes instead of actually comparing a dictionary here.

Therefore, not using extraction logic from Context Propagaiton.